File: src/service/ReturnOrderService.js
import service from './api';

class ReturnOrderService {

    get(orderId) {
        return service.post(`/${orderId}/return-orders/get-by-order`, { id: orderId });
    }

    create(orderId, data) {
        return service.post(`/${orderId}/return-orders`, data);
    }

    update(orderId, id, data) {
        return service.put(`/${orderId}/return-orders/${id}`, data);
    }

    delete(orderId, id) {
        return service.delete(`/${orderId}/return-orders/${id}`);
    }

}

export default new ReturnOrderService();

File: back-end/vrp-preprocess/src/main/java/com/vrp/demo/service/ReturnOrderService.java
package com.vrp.demo.service;

import com.vrp.demo.entity.tenant.ReturnOrder;
import com.vrp.demo.exception.CustomException;
import com.vrp.demo.models.ReturnOrderModel;
import com.vrp.demo.models.search.OrderSearch;

public interface ReturnOrderService extends BaseService<ReturnOrder, Long>{

    public ReturnOrderModel find(OrderSearch search);

    public ReturnOrderModel create(ReturnOrderModel returnOrderModel) throws CustomException;

    public ReturnOrderModel update(ReturnOrderModel returnOrderModel) throws CustomException;

    public int delete(Long id) throws CustomException;

}
